src/server/components/zitadel/utils.ts

```diff
@@ -0,0 +1,183 @@
+import type {Request} from '@gravity-ui/expresskit';
+import type {AppContext} from '@gravity-ui/nodekit';
+import axios from 'axios';
+import NodeCache from 'node-cache';
+
+import {
+    clientId,
+    clientSecret,
+    serviceClientId,
+    serviceClientSecret,
+    zitadelProjectId,
+    zitadelUri,
+} from '../../app-env';
+import {getDuration} from '../charts-engine/components/utils';
+
+const cache = new NodeCache();
+
+export const introspect = async (ctx: AppContext, token?: string): Promise<boolean> => {
+    ctx.log('Token introspection');
+
+    if (!zitadelUri) {
+        throw new Error('Missing ZITADEL_URI in env');
+    }
+    if (!clientId) {
+        throw new Error('Missing CLIENT_ID in env');
+    }
+    if (!clientSecret) {
+        throw new Error('Missing CLIENT_SECRET in env');
+    }
+
+    try {
+        if (!token) {
+            throw new Error('Token not provided');
+        }
+
+        const hrStart = process.hrtime();
+
+        const axiosInstance = axios.create();
+
+        const response = await axiosInstance({
+            method: 'post',
+            url: `${zitadelUri}/oauth/v2/introspect`,
+            auth: {
+                username: clientId,
+                password: clientSecret,
+            },
+            'axios-retry': {
```

Comment:
Will it retry without calling `axiosRetry(axiosInstance);` call?

Reply:
you are right, we need to use external interceptor